File: cplusplus/design_pattern/VisitorProj/ConcreteNode1.cpp
#include "stdafx.h"
#include "ConcreteNode1.h"
#include "Visitor.h"

ConcreteNode1::ConcreteNode1()
{
}


ConcreteNode1::~ConcreteNode1()
{
}

void ConcreteNode1::showMe()
{
    std::cout << "ConcreteNode1::showMe()" << std::endl;
}

void ConcreteNode1::accept(std::shared_ptr<Visitor> v)
{
    v->visit(shared_from_this());
}

File: Java/pattern/VisitorProj/src/com/pattern/ConcreteNode1.java
package com.pattern;

public class ConcreteNode1 implements Node {

	@Override
	public void showMe() {
		System.out.println("ConcreteNode1::showMe()");
	}

	@Override
	public void accept(Visitor v) {
		v.visit(this);
	}
	

}
